definitions/npm/styled-components_v2.x.x/flow_v0.53.x-/styled-components_v2.x.x.js

```diff
@@ -0,0 +1,332 @@
+// @flow
+
+type $npm$styledComponents$Interpolation = ((executionContext: Object) => string) | string | number;
+type $npm$styledComponents$NameGenerator = (hash: number) => string;
+
+type $npm$styledComponents$TaggedTemplateLiteral<R> = {| (Array<string>, $npm$styledComponents$Interpolation): R |};
+
+type $npm$styledComponents$ReactComponentFunctional<Props: {}> = Props => React$Element<*>;
+type $npm$styledComponents$ReactComponentClass<Props, DefaultProps = *> = Class<React$Component<DefaultProps & Props, *>>
+
+type $npm$styledComponents$ReactComponentUnion<Props> =
```

Comment:
Is there a reason not to use `React$ComponentType` in `flow@0.53`?

Reply:
``React$ComponentType`` has a ``Props`` type parameter, but your return type omits it.
```js
declare function $npm$styledComponents$WithTheme<Props: {}>(
Component: React$ComponentType<{ theme: $npm$styledComponents$Theme & Props}>
) : React$ComponentType
```

Reply:
As of right now, I'm typing ``withTheme`` as a morphism of both ``Props`` and ``DefaultProps``. If a component requires a theme prop, ``withTheme`` removes the need to provide it when instantiating (``<Component />``) & "adds" (intersection) the prop to the ``DefaultProps`` of the component.

Reply:
```js
declare var needsComponentWithDefaultProp: Class<React$Component<{}, { theme: $npm$styledComponents$Theme }>> => void

class NeedsThemePropComponent extends React.Component<{ theme: $npm$styledComponents$Theme }, {}> {}

const Foo = withTheme(NeedsThemePropComponent)

needsComponentWithDefaultProp(Foo) // OK
<Foo /> // OK
```